plugins/animate/animate.cpp

```diff
@@ -112,8 +122,29 @@ struct animation_hook : public animation_hook_base
             view->set_minimized(true);
         }
 
-        /* Will also delete this */
-        view->erase_data(animate_custom_data_id);
+        if (type & MAP_STATE_ANIMATION)
```

Comment:
Why do we need this differentiation, isn't the correct name passed to this struct already in the constructor?